arff.py

```diff
@@ -839,24 +875,28 @@ def iter_encode(self, obj):
         # ATTRIBUTES
         if not obj.get('attributes'):
             raise BadObject('Attributes not found.')
-            
+
         for attr in obj['attributes']:
             # Verify for bad object format
-            if not isinstance(attr, (tuple, list)) or \
-               len(attr) != 2 or \
-               not isinstance(attr[0], basestring):
-                raise BadObject('Invalid attribute declaration "%s"'%str(attr))
+            if not isinstance(attr, tuple):
+                if not isinstance(attr, list):
+                    raise BadObject('Invalid attribute')
+            # Verify for invalid types
+            if not isinstance(attr[1], list) and attr[1] not in _SIMPLE_TYPES:
+                raise BadObject('Invalid attribute type "%s"' % str(attr))
 
-            if isinstance(attr[1], basestring):
-                # Verify for invalid types
-                if attr[1] not in _SIMPLE_TYPES:
-                    raise BadObject('Invalid attribute type "%s"'%str(attr))
+            if not isinstance(attr, (tuple, list)) or \
+                    not isinstance(attr[0], basestring):
+                raise BadObject('Invalid attribute declaration "%s"' % str(attr))
 
+            if isinstance(attr[1], basestring) and attr[1].upper() == "DATE" and len(attr) == 3:
+                yield self._encode_attribute(attr[0], attr[1], attr[2])
```

Comment:
Could you please try to re-arrange the code such that the yield statements are next to each other? This should improve readability, especially as I was wondering 'what happens if the length of the attributes is != 2. Maybe you could add an else statement here just in case.